lib/Cake/Test/Case/Controller/Component/CookieComponentTest.php

```diff
@@ -790,9 +752,99 @@ public function testDeleteChildrenNotExist() {
 	}
 
 /**
+ * Test that deleting a child entry of the cookie array works
+ *
+ * @return void
+ */
+	public function testDeleteChildEntry() {
+		$response = $this->Controller->response;
+
+		$this->Cookie->write('User.name', 'user', false);
+		$this->Cookie->write('User.email', 'user@example.com', false);
```

Comment:
Now I understand that I must not revert the 07f4779efef507c31291ca0f29222c1b06d626b8. Going from secure to insecure is never acceptable. I must un-revert it.

However, I would like to make my test cases passed. What if we allow people to set how cookies are sent to the clients?

If `useSingleCookie` is set to `true` (compatible with the versions greater than 2.4.3), it would be better to raise an error or a warning if people try to call `write()` with the third and/or fourth arguments for the deeply nested values:
```php
$this->Cookie->write('foo.bar', 'value', true, '+1 day'); // Error
```
And If it is set to `false` (compatible with the versions earlier than 2.4.3), it allows to call `write()` with those arguments for the deeply nested values.

It defaults to `true`, as the latest 2.x versions  work so.